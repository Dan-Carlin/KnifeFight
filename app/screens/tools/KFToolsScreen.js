import React, { useState } from "react";
import { View, StyleSheet, BackHandler } from "react-native";
import { useFocusEffect } from "@react-navigation/native";

import {
  ArrowLeftGraphic,
  ArrowRightGraphic,
  BannerGraphic,
  DiceRollerGraphic,
  MinusGraphic,
  PlusGraphic,
} from "../../assets/buttons/actions";
import Button from "../../components/Button";
import colors from "../../config/colors";
import CounterContainer from "../../components/CounterContainer";
import CustomAlert from "../../components/CustomAlert";
import { HomeGraphic } from "../../assets/buttons/navigation";
import InstructionsModal from "../../components/InstructionsModal";
import Modal from "../../components/Modal";
import NameDisplay from "../../components/NameDisplay";
import OpacityButton from "../../components/OpacityButton";
import routes from "../../navigation/routes";
import Screen from "../../components/Screen";
import SoundButton from "../../components/SoundButton";
import Text from "../../components/Text";
import * as TraitSvgs from "../../assets/traits";

const background = require("../../assets/backgrounds/kf_background_land_xxxhdpi.png");

function KFToolsScreen({ navigation }) {
  const [exitModalVisible, setExitModalVisible] = useState(false);
  const [howToModalVisible, setHowToModalVisible] = useState(true);

  const onBackPress = () => {
    setExitModalVisible(true);
    return true;
  };

  useFocusEffect(
    React.useCallback(() => {
      BackHandler.addEventListener("hardwareBackPress", onBackPress);

      return () =>
        BackHandler.removeEventListener("hardwareBackPress", onBackPress);
    }, [])
  );

  return (
    <Screen style={styles.screenContainer} background={background}>
      <Modal
        component={
          <InstructionsModal onConfirm={() => setHowToModalVisible(false)} />
        }
        isVisible={howToModalVisible}
        setIsVisible={() => setHowToModalVisible(!howToModalVisible)}
      />

      <Modal
        component={
          <CustomAlert
            onCancel={() => setExitModalVisible(false)}
            onConfirm={() => navigation.popToTop()}
          />
        }
        isVisible={exitModalVisible}
        setIsVisible={() => setExitModalVisible(!exitModalVisible)}
      />

      <View style={styles.leftContainer}>
        <View style={styles.topButtons}>
          <SoundButton style={styles.soundButton} />
        </View>
        <View style={styles.diceRollButton}>
          <OpacityButton
            Graphic={DiceRollerGraphic}
            onPress={() => console.log("Dice roller button pressed.")}
          />
        </View>
      </View>
      <View style={styles.centerContainer}>
        <View style={styles.nameContainer}>
          <View style={styles.styleEditContainer}>
            <View style={styles.styleButtons}>
              <Button
                Graphic={ArrowLeftGraphic}
                onPress={() => console.log("Previous style pressed.")}
              />
            </View>
            <View style={styles.fontPickerContainer}>
              <Text style={styles.styleLabel}>Font Style:</Text>
              <Text style={styles.currentFont}>Default</Text>
            </View>
            <View style={styles.styleButtons}>
              <Button
                Graphic={ArrowRightGraphic}
                onPress={() => console.log("Next style pressed.")}
              />
            </View>
          </View>
          <View style={styles.nameDisplayContainer}>
            <NameDisplay
              gangName={"Gang Name"}
              gangTrait={"Unstable"}
              font="default"
              color={colors.gang_red}
            />
          </View>
        </View>
        <View style={styles.bottomContainer}>
          <View style={styles.hpButtons}>
            <Button
              Graphic={MinusGraphic}
              onPress={() => console.log("Minus button pressed.")}
            />
          </View>
          <CounterContainer
            width={240}
            Graphic={TraitSvgs.Unstable}
            hpValue="25"
            gangColor={colors.gang_red}
          />
          <View style={styles.hpButtons}>
            <Button
              Graphic={PlusGraphic}
              onPress={() => console.log("Plus button pressed.")}
            />
          </View>
        </View>
      </View>
      <View style={styles.rightContainer}>
        <View style={styles.topButtons}>
          <OpacityButton Graphic={HomeGraphic} onPress={() => onBackPress()} />
        </View>
        <View style={styles.bannerButton}>
          <OpacityButton
            Graphic={BannerGraphic}
            onPress={() => navigation.navigate(routes.BANNER)}
          />
        </View>
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  bannerButton: {
    height: "50%",
    padding: 10,
    width: "100%",
  },
  bottomContainer: {
    alignItems: "center",
    flex: 0.5,
    flexDirection: "row",
    paddingHorizontal: 10,
    width: "100%",
  },
  centerContainer: {
    alignItems: "center",
    flex: 0.6,
  },
  currentFont: {
    fontSize: 18,
    fontStyle: "italic",
    fontWeight: "bold",
    textAlign: "center",
  },
  diceRollButton: {
    height: "50%",
    padding: 30,
    width: "100%",
  },
  fontPickerContainer: {
    alignItems: "center",
    height: "100%",
    justifyContent: "center",
    width: "30%",
  },
  hpButtons: {
    bottom: -25,
    flex: 1,
    margin: 10,
  },
  leftContainer: {
    alignItems: "center",
    flex: 0.2,
    justifyContent: "space-between",
  },
  nameContainer: {
    alignItems: "center",
    flex: 0.5,
    width: "100%",
  },
  nameDisplayContainer: {
    flex: 0.65,
    height: "100%",
    width: "100%",
  },
  rightContainer: {
    alignItems: "center",
    flex: 0.2,
    justifyContent: "space-between",
  },
  screenContainer: {
    flex: 1,
    flexDirection: "row",
  },
  soundButton: {
    height: "60%",
    width: "60%",
  },
  styleButtons: {
    height: "100%",
    padding: 14,
    width: "20%",
  },
  styleEditContainer: {
    alignItems: "center",
    flex: 0.35,
    flexDirection: "row",
    justifyContent: "center",
    width: "100%",
  },
  styleLabel: {
    color: colors.medium,
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "center",
  },
  topButtons: {
    alignItems: "center",
    flexDirection: "row",
    height: "25%",
    justifyContent: "center",
    padding: 15,
    width: "100%",
  },
});

export default KFToolsScreen;
