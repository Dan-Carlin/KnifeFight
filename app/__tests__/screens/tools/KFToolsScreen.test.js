import React from "react";
import renderer from "react-test-renderer";

import KFToolsScreen from "../../../screens/tools/KFToolsScreen";

jest.useFakeTimers();

describe("Tools screen tests", () => {
  it("renders Tools screen correctly", () => {
    const tree = renderer.create(<KFToolsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // decrements HP value when minus button is tapped

  // increments HP value when plus button is tapped

  // sets gang name correctly

  // sets gang name color correctly

  // sets HP label color correctly

  // sets Trait svg correctly
});
