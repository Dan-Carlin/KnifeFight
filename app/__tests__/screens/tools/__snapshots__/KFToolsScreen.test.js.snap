// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools screen tests renders Tools screen correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "paddingTop": 20,
        },
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
  >
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={true}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#000000",
              "opacity": 0.3,
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#414042",
              "borderColor": "#000",
              "borderRadius": 20,
              "height": "165%",
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "rotate": "-90deg",
                },
              ],
              "width": "45%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0005",
                "borderBottomColor": "#000",
                "borderEndColor": "#000",
                "borderStartColor": "#bcbec0",
                "borderTopColor": "#bcbec0",
                "borderTopEndRadius": 20,
                "borderTopStartRadius": 20,
                "borderWidth": 2,
                "flex": 0.06,
                "flexDirection": "row",
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  Object {
                    "fontFamily": "default",
                    "fontSize": 40,
                    "textAlign": "center",
                  },
                ]
              }
            >
              How to use this app:
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 0.83,
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 0.85,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginEnd": 6,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontFamily": "Avenir",
                            "fontSize": 16,
                            "width": "100%",
                          },
                          undefined,
                        ]
                      }
                    >
                      1.
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexWrap": "wrap",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontFamily": "Avenir",
                            "fontSize": 16,
                            "width": "100%",
                          },
                          undefined,
                        ]
                      }
                    >
                      Set down your phone on the play area in front of you as shown.
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "height": "60%",
                      "justifyContent": "flex-start",
                      "padding": 10,
                    }
                  }
                >
                  <SvgMock />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.15,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": "80%",
                      "width": "25%",
                    }
                  }
                >
                  <SvgMock
                    style={
                      Object {
                        "opacity": 0.5,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontFamily": "Avenir",
                          "fontSize": 16,
                          "width": "100%",
                        },
                        Object {
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    1/3
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "height": "80%",
                      "width": "25%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "height": "100%",
                        "opacity": 1,
                        "width": "100%",
                      }
                    }
                  >
                    <SvgMock />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.11,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "height": "60%",
                  "width": "50%",
                }
              }
            >
              <SvgMock
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              />
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#000000",
              "opacity": 0.3,
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#414042",
                "borderColor": "#000",
                "borderRadius": 20,
                "borderWidth": 2,
                "height": "50%",
                "overflow": "hidden",
                "width": "60%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#0005",
                  "borderBottomColor": "#000",
                  "borderEndColor": "#000",
                  "borderStartColor": "#bcbec0",
                  "borderTopColor": "#bcbec0",
                  "borderTopEndRadius": 20,
                  "borderTopStartRadius": 20,
                  "borderWidth": 2,
                  "flex": 0.25,
                  "flexDirection": "row",
                  "padding": 4,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": "100%",
                    "marginEnd": 6,
                    "width": "10%",
                  }
                }
              >
                <SvgMock />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "width": "100%",
                    },
                    Object {
                      "fontFamily": "default",
                      "fontSize": 50,
                    },
                  ]
                }
              >
                Hold up!
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.4,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "width": "100%",
                    },
                    Object {
                      "color": "#bcbec0",
                    },
                  ]
                }
              >
                Are you sure you want to end the game? 
(All data will be lost.)
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.35,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": "80%",
                    "width": "30%",
                  }
                }
              >
                <SvgMock
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                />
              </View>
              <View
                style={
                  Object {
                    "height": "80%",
                    "width": "30%",
                  }
                }
              >
                <SvgMock
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "25%",
            "justifyContent": "center",
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
              Object {
                "height": "60%",
                "width": "60%",
              },
            ]
          }
        >
          <SvgMock
            style={
              Object {
                "height": "100%",
                "width": "35%",
              }
            }
          />
          <SvgMock
            style={
              Object {
                "height": "100%",
                "width": "35%",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "50%",
            "padding": 30,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.5,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.35,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "height": "100%",
                "padding": 14,
                "width": "20%",
              }
            }
          >
            <SvgMock
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  Object {
                    "color": "#7a7b7c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Font Style:
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  Object {
                    "fontSize": 18,
                    "fontStyle": "italic",
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Default
            </Text>
          </View>
          <View
            style={
              Object {
                "height": "100%",
                "padding": 14,
                "width": "20%",
              }
            }
          >
            <SvgMock
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 0.65,
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              mode="max_lines"
              numberOfLines={1}
              onTextLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "color": "#9e9fa1",
                    "fontFamily": "default",
                    "textAlign": "center",
                    "width": "94%",
                  },
                  Object {
                    "fontSize": 200,
                  },
                ]
              }
              testID="max-lines"
            >
              ???
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "bottom": -25,
              "flex": 1,
              "margin": 10,
            }
          }
        >
          <SvgMock
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#414042",
              "borderColor": "#bcbec0",
              "borderEndWidth": 6,
              "borderStartWidth": 6,
              "borderTopEndRadius": 120,
              "borderTopStartRadius": 120,
              "borderTopWidth": 6,
              "height": "100%",
              "justifyContent": "center",
              "padding": 20,
              "width": 240,
            }
          }
        >
          <SvgMock
            style={
              Object {
                "bottom": 8,
                "opacity": 0.7,
                "position": "absolute",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "bottom": -20,
                  "color": "#9e9fa1",
                  "fontFamily": "default",
                  "fontSize": 100,
                  "textAlign": "center",
                },
              ]
            }
          >
            HP
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "bottom": 5,
                  "fontSize": 100,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            25
          </Text>
        </View>
        <View
          style={
            Object {
              "bottom": -25,
              "flex": 1,
              "margin": 10,
            }
          }
        >
          <SvgMock
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "25%",
            "justifyContent": "center",
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "50%",
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
